### DISCUSSION TITLE
Branches

### DISCUSSION THREAD

**Discussion by ricoderks** [ndcn/omicser#173] [Ideas]
@ergonyc is it maybe an idea to setup a separate branch for development and leave the main branch as the branch which has a version number and is always working?

Cheers, Rico

**Comment by ergonyc** [comment]
@ricoderks Absolutely!   We should decide:

1. what its called... e.g. `main-dev`?  I can't recall a github standard off the top of my head...?
2. criteria for _minor_ iterations of the "release" version.  i.e. when the `main-dev` branch gets sufficiently tested and merges into main. 

The nascent WIP "roadmap" will map out the _major_ releases... i.e. 2.0 which has "linking" datasets capabilities... so we just need to feel good about what _minor_ functionality we will iterate.

**Reply by ricoderks** [reply]
@ergonyc 

1. I think there often called `dev` or `main-dev`. I'am also not sure if there is a github standard. 
2. Yes, for sure. I have no experience with this. I'll have a look what kind of criteria's are normally used.

**Reply by ricoderks** [reply]
Looking into no. 2, I found the simplest is x.y.z

* x is major release
    * This is linked to our roadmap.
* y is minor release
    * Small feature update(s).
    * Important not to break backward compatibility.
* z is bug fixes

To avoid that we release a minor update too often. I would say we collect a few feature updates before we release. Except, if the last minor release is a while ago or the feature request was urgent.

Cheers, Rico

**Reply by ergonyc** [reply]
@ricoderks 
I think we should go ahead and create a `main-dev` branch.   Should we also start a discussion on _testing_?  I hadn't implemented any unit testing, but as things get more it will probably be nescessary... and it _IS_ best practices for a reason.

-----------

Rough Roadmap: (TODO: publish ROADMAP.md)
- v1.0 (current) - basic browser (`main` branch)
   - v1.1 (export functions) (to be merged `exporter` branch)
  - implement unit testing (first as `testing` branch, then as part of `main-dev`)

- v2.0 (as soon as late Feb) - cross-dataset linking functionality (`main-dev` branch)


- v3.0 (future) - integrative / harmonized cross-omics functionality

**Reply by ricoderks** [reply]
Sorry for my late reaction, me and my family got hit by covid and are still recovering a bit!

I think it's for sure a good idea to also start with testing. In the beginning it is maybe a bit more work, but can save us a lot of work in the long run. 

`main-dev` sounds good! In there we can make maybe different branches for different functionalities we want to add.
